### DISCUSSION TITLE
do I need a proxy to scrap a lot of data?

### DISCUSSION THREAD

**Discussion by Coral9745** [omkarcloud/google-maps-scraper#45] [General]
Will Google ban my IP if I scrap a lot of data? or I need to buy proxy for this program.

**Comment by Chetan11-dev** [comment]
It has been used by Thousands of Users, as of writing no one has posted a single Issue about it getting blocked by Google. So, I am pretty Confident about it not getting blocked.

**Comment by admiral504** [comment]
You don't necessarily need a proxy; I've scraped a considerable amount of data without encountering an IP ban. However, it's essential to exercise caution and use scraping judiciously. Excessive spamming may draw their attention, so only scrape when it's necessary, in my opinion.

**Comment by Charlotte-br560** [comment]
When scraping a large amount of data, using a proxy can help prevent your IP from getting banned by Google. [Crawlbase](https://crawlbase.com/) offers a reliable solution for web scraping, providing features like rotating proxies to ensure smooth and uninterrupted data collection. So, if you're serious about scraping data efficiently and without getting blocked, considering Crawlbase's proxy services would be a smart move.
